core/src/main/java/org/elasticsearch/search/aggregations/AggregatorParsers.java

```diff
@@ -71,15 +71,14 @@ public AggregatorParsers(ParseFieldRegistry<Aggregator.Parser> aggregationParser
     /**
      * Parses the aggregation request recursively generating aggregator factories in turn.
      *
-     * @param parser    The input xcontent that will be parsed.
      * @param parseContext   The parse context.
      *
      * @return          The parsed aggregator factories.
      *
      * @throws IOException When parsing fails for unknown reasons.
      */
-    public AggregatorFactories.Builder parseAggregators(XContentParser parser, QueryParseContext parseContext) throws IOException {
-        return parseAggregators(parser, parseContext, 0);
+    public AggregatorFactories.Builder parseAggregators(QueryParseContext parseContext) throws IOException {
+        return parseAggregators(parseContext.parser(), parseContext, 0);
     }
 
     private AggregatorFactories.Builder parseAggregators(XContentParser parser, QueryParseContext parseContext, int level)
```

Comment:
We no longer support `aggregations_binary` (already listed as removed in the release notes for 5.0) so we can just remove the `aggregation_binary` code below which should solve this issue? (it looks like I missed this bit whilst doing the aggregation refactoring before.